daemon/alloc_hooks.def

```diff
@@ -32,3 +32,5 @@ X(release_free_memory, void, ())
 X(enable_thread_cache, bool, (bool enable))
 X(get_allocator_property, bool, (const char* name, size_t* value))
 X(set_allocator_property, bool, (const char* name, size_t value))
+X(set_allocator_arena, bool, (size_t arenaid))
+X(get_arena_allocation_size, size_t, (size_t arenaid))
```

Comment:
I'm not sure if we want to add two new methods for these, or simply use the existing get/set_allocator_property APIs - while jemalloc is the only "production" allocator we currently use, we do run tests with the system allocator and badmalloc - and hence it would confuse the interface if we add a bunch of methods specific to jemalloc, when we can just use the generic method.